master.js

```diff
@@ -440,37 +440,60 @@ app.post("/api/uploadMod", authenticate.middleware, function(req,res) {
     }
 });
 /**
+Prepare the slaveCache for both /api/slaves calls
+*/
+let slaveCache = {
+	timestamp: Date.now(),
+};
+function getSlaves() {
+	if(!slaveCache.cache || Date.now() - slaveCache.timestamp > 5000){
+		let copyOfSlaves = JSON.parse(JSON.stringify(slaves));
+		slaveCache.cache = copyOfSlaves;
+		slaveCache.timestamp = Date.now();
+	}
+	return slaveCache;
+}
+/**
 GET endpoint for getting information about all our slaves
 @memberof clusterioMaster
 @instance
 @alias /api/slaves
 */
-let slaveCache = {
-	timestamp: Date.now(),
-};
 app.get("/api/slaves", function(req, res) {
 	endpointHitCounter.labels(req.route.path).inc();
 	res.header("Access-Control-Allow-Origin", "*");
 	res.header("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept");
-	if(!slaveCache.cache || Date.now() - slaveCache.timestamp > 5000){
-		let copyOfSlaves = JSON.parse(JSON.stringify(slaves));
-		// filter out the rcon password because thats kindof not a safe thing to share
-		for(key in copyOfSlaves){
-			copyOfSlaves[key].rconPassword = "hidden";
-		}
-		slaveCache.cache = copyOfSlaves;
-		slaveCache.timestamp = Date.now();
+	let slaveCache = getSlaves();
+	for(key in slaveCache) {
+		slaveCache[key].rconPassword = "hidden";
 	}
-	
 	res.send(slaveCache.cache);
 });
-var recievedItemStatisticsBySlaveID = {};
-var sentItemStatisticsBySlaveID = {};
+/**
+POST endpoint for getting information about all our slaves, requires auth, responds including RCON passwords
+@memberof clusterioMaster
+@instance
+@alias /api/slaves
+*/
+app.post("/api/slaves", function(req, res) {
+	let token = req.headers['x-access-token'];
+	if(!token) return res.status(401).send({ auth: false, message: 'No token provided.' });
+	jwt.verify(token, config.masterAuthSecret, function(err, decoded) {
+		if(err) return res.status(500).send({ auth: false, message: 'Failed to authenticate token.' });
```

Comment:
I don't want to be an ass, but I honestly always require full block in any code review I do (and it's a forced transformation setting in my IDE)
```
if (condition) {
    // do stuff
}
```

Practice and experience show that non-block structure will get messed up or fucked with a 99% probability and will lead to bugs.

It also took me a while to parse it visually and see what is exactly executed in the condition.
Also, the line is just long :)

Could we please edit it?